src/core/lib/surface/call.cc

```diff
@@ -157,6 +158,7 @@ struct grpc_call {
   gpr_timespec start_time = gpr_now(GPR_CLOCK_MONOTONIC);
   /* parent_call* */ gpr_atm parent_call_atm = 0;
   child_call* child = nullptr;
+  grpc_core::Atomic<intptr_t> batches_completed;
```

Comment:
Given that this still needs to be an atomic, I'm a little surprised that this actually improves performance -- we're basically just updating this atomic instead of the refcount atomic.  Is the savings just because this atomic can use `MemoryOrder::RELAXED` and the refcount atomic can't?

Reply:
Two-fold reason: RELAXED semantics, and no branch on Ref() (while Unref() has a branch to see if the Ref is now 0).